apps/src/lib/kits/maker/ui/SetupChecklist.jsx

```diff
@@ -350,8 +407,22 @@ export default class SetupChecklist extends Component {
               </div>
             )}
             {!linuxPermissionError && this.installFirmwareSketch()}
-            {this.contactSupport()}
+            {experiments.isEnabled('microbit') &&
+              this.state.boardTypeDetected === BOARD_TYPE.MICROBIT &&
```

Comment:
I might extract these conditions to a separate constant at the top of the `render` method to clarify what they're doing. 

Ex. 
```
const firmataNeedsUpdate = this.state[STATUS_BOARD_CONNECT] === Status.FAILED && (this.state[STATUS_BOARD_UPDATE_FIRMATA] === Status.WAITING ||
                this.state[STATUS_BOARD_UPDATE_FIRMATA] === Status.FAILED);
```

And then...
```
firmataNeedsUpdate && <Button .../>
```

I'm not sure `firmataNeedsUpdate` is the right name here, though. We want the `Update micro:bit` button to appear even if the board update has already failed once?

Reply:
That's a great suggestion - will do. Thanks!

Reply:
And yes, we want the 'Update micro:bit' button to appear if Firmata update failed. For example, if the user disconnects the micro:bit before flashing is completed.